app/models/custom_button_set.rb

```diff
@@ -1,6 +1,19 @@
 class CustomButtonSet < ApplicationRecord
   acts_as_miq_set
 
+  before_save :update_children
+  def update_children
+    if set_data.nil? || set_data[:button_order].nil?
+      remove_all_children
+      return
+    end
+
+    children = Rbac.filtered(CustomButton.where(:id => set_data[:button_order]))
+    throw(:abort) if children.pluck(:id).sort != set_data[:button_order].sort
```

Comment:
I think you want to use `raise` here with an actual exception and its explanation.

Reply:
Well, as far as I can tell, `throw :abort` is the only way for `before_save` to abort that's compatible with rails 5.

([docs](https://guides.rubyonrails.org/active_record_callbacks.html#halting-execution))